### PR TITLE
Homework #5.

### PR DESCRIPTION
**Pull request by interist32**
Logging with Winston.

### CODE REVIEW

File: hw-5/server.ts [hunk 1]
```diff
@@ -7,14 +7,29 @@ import userRouter from './routers/user.router';
...
-  console.log(`Server is listening on http://localhost:${PORT}`);
+  logger.info(`Server is listening on http://localhost:${PORT}`);
 });
+
+process.on('unhandledRejection', function(reason) {
+  logger.error('unhandledRejection', reason);
+});
+
+process.on('uncaughtException', function(err) {
+  logger.error('UncaughtException:', err.message);
```
Comment (aledpardo) [inline hw-5/server.ts]: I believe the `:` character is not needed.

File: hw-5/middlewares/api-logger.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import {NextFunction, Request, Response} from 'express';
+import {logger} from './logger';
+
+export const apiLogger = (req: Request, res: Response, next: NextFunction) => {
+  let logStr = `[${req.method}] ${req.url}\n Params: ${
+      JSON.stringify(req.params)}\n Body: ${JSON.stringify(req.body)}`;
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-5/middlewares/api-logger.ts]: Makes sense! Thanks for an explanation. Fixed.

File: hw-5/middlewares/error-handler.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import {NextFunction, Request, Response} from 'express';
+
+export const errorHandler =
+    (err: Error, req: Request, res: Response, next: NextFunction) => {
+      if (res.headersSent) {
+        return next(err);
+      }
+      res.status(500);
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-5/middlewares/error-handler.ts]: Done!

File: hw-5/server.ts (hunk 1 above)
Comment (interist32) [reply hw-5/server.ts]: Right! Thanks!

**Pull request merged**